### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

**Comment by sungam3r** [comment]
https://app.codecov.io/gh/graphql-dotnet/parser
изображение

File: src/GraphQLParser/AST/GraphQLLocation.cs [hunk 1]
```diff
@@ -1,12 +1,10 @@
 using System;
-using System.Diagnostics;
 
 namespace GraphQLParser.AST;
 
 /// <summary>
 /// Provides information regarding the location of a node within a document's original text.
 /// </summary>
-[DebuggerDisplay("(Start={Start}, End={End})")]
```
Comment (sungam3r) [inline src/GraphQLParser/AST/GraphQLLocation.cs:9]: Not needed, `ToString()` exists.

**Comment by sungam3r** [comment]
изображение

**Comment by Shane32** [comment]
Nice!

**Comment by sungam3r** [comment]
I doubt that some areas can be covered with tests. Like this one
изображение

**Comment by Shane32** [comment]
It doesn’t count consistently either. You might find it counts it as 100%. But the hardest part is writing tests for error checks on private methods (which shouldn’t occur) or values returned by external methods which should never return an error. Mocks help but can’t always simulate the issue if the external method is part of a sealed class.

**Comment by Shane32** [comment]
For my recent small projects built with 100% testing in mind, eventually it seems a bit pointless as you get much above 90%. On my corporate code where I recently started a new project “tests-first” I’m aiming for at least 80%. But aiming for a higher percentage increasingly becomes less productive with little value.

**Comment by Shane32** [comment]
Like all the serializers work I’m doing, I could simulate invalid json syntax for every type being deserialized, hitting every bit of “throw” code…. But I can’t simulate the json parser returning a value node where it expects a property node.

**Comment by Shane32** [comment]
It is really satisfying to see the coverage at 98%+ though!

**Comment by sungam3r** [comment]
As I was convinced, it is important not so much the value of the coverage, how much the process of finding errors while increasing this coverage.

**Comment by Shane32** [comment]
For sure. I’ve found so many bugs while attempting to increase coverage.

**Pull request merged**